### PR TITLE
Improve user creation and editing in SDI

### PR DESCRIPTION
**Pull request by 2e2a**
Consists of the following changes:
* Make password optional when creating a user by generating a default password
* Create new ActivationConfigutation sheet for the user resource, to control if a newly created user is directly activated of if a notification or invitation mail is send.

### CODE REVIEW

File: src/adhocracy_core/adhocracy_core/schema/test_init.py [hunk 1]
```diff
@@ -710,6 +716,18 @@ def test_deserialize_non_valid_no_str(self):
...
             inst.deserialize(1)
 
+    def test_bind_and_serialize_empty(self):
+        from datetime import datetime
+        inst = self.make_one().bind()
+        result = inst.serialize()
+        assert len(result) == 20
+
+    def test_bind_and_setup_datetime_widget(self):
+        from deform.widget import PasswordWidget
```
Comment (joka) [inline src/adhocracy_core/adhocracy_core/schema/test_init.py]: you mean password widget

File: src/adhocracy_core/adhocracy_core/sheets/principal.py [hunk 2]
```diff
@@ -324,6 +326,48 @@ def check_plaintext_password(self, password: str) -> bool:
...
+    activation = ActivationSetting()
+
+
+activation_configuration_meta = sheet_meta._replace(
+    isheet=IActivationConfiguration,
+    schema_class=ActivationConfigutationSchema,
+    readable=False,
+    creatable=True,
+    editable=False,
+    permission_create='create_user',
```
Comment (joka) [inline src/adhocracy_core/adhocracy_core/sheets/principal.py]: We need a special permission for admin, we don't want that every users can do this

File: src/adhocracy_core/adhocracy_core/sheets/principal.py [hunk 3]
```diff
@@ -324,6 +326,48 @@ def check_plaintext_password(self, password: str) -> bool:
...
+    """
+
+    activation = ActivationSetting()
+
+
+activation_configuration_meta = sheet_meta._replace(
+    isheet=IActivationConfiguration,
+    schema_class=ActivationConfigutationSchema,
+    readable=False,
+    creatable=True,
```
Comment (joka) [inline src/adhocracy_core/adhocracy_core/sheets/principal.py]: maybe readable is usefull, if you want to see later how the user was created/activated.

File: src/adhocracy_core/adhocracy_core/resources/subscriber.py [hunk 4]
```diff
@@ -244,21 +245,35 @@ def send_password_reset_mail(event):
...
     """
-    settings = event.registry.settings
     user = event.object
-    skip_mail = asbool(settings.get('adhocracy.skip_registration_mail', False))
-    if skip_mail:
+    registry = event.registry
+    sheet = registry.content.get_sheet(user, IActivationConfiguration)
+    activation_config = sheet.get()['activation']
+    if _is_activation_disabled(registry):
         user.activate()
```
Comment (joka) [inline src/adhocracy_core/adhocracy_core/resources/subscriber.py]: skip_registration_mail was only a workaround for thentos. It makes more sense to add this option to the activationConfiguration sheet, then we can also add users with the sdi without activation mail.

File: src/adhocracy_core/adhocracy_core/schema/__init__.py [hunk 5]
```diff
@@ -687,9 +701,10 @@ class Password(SchemaNode):
     """
 
     schema_type = StringType
-    default = ''
+    default = deferred_password_default
     missing = drop
```
Comment (joka) [inline src/adhocracy_core/adhocracy_core/schema/__init__.py]: nice

File: src/adhocracy_core/adhocracy_core/sheets/principal.py (hunk 2 above)
Comment (joka) [inline src/adhocracy_core/adhocracy_core/sheets/principal.py]: migration to set this sheet is missing

**Comment by 2e2a** [comment]
comments applied

**Comment by joka** [comment]
cool

**Pull request merged**